### ISSUE TITLE
Orthography replacement prevents undo stroke from functioning properly in space after output mode.

### ISSUE THREAD

**Issue by jamaclellan** [openstenoproject/plover#961]
## Summary
When operating in space after output mode Plover will lose track of the previous state when undoing a stroke that includes an orthography replacement rule, such as `made` -> `madies`. This only occurs when a letter is replaced from the original word.

## Reproducing
Set plover space placement to "After Output".
Stroke `PHAED`
Stroke `KWREUS`
Stroke `*`
Output is displayed as `mad ` instead of `made `

## Plover Version
Plover 4.0.0.dev8+8.ge062cda 

## System
Windows 10 1709

**Comment by morinted** [comment]
Blackbox Test

```diff
diff --git a/test/test_blackbox.py b/test/test_blackbox.py
index 731c5d9..e15839b 100644
--- a/test/test_blackbox.py
+++ b/test/test_blackbox.py
@@ -451,6 +451,16 @@ class TestsBlackbox(BlackboxTester):
         TPAOEUPL/-G  ' TIMing'
         '''

+    def test_orthography_issue_961(self):
+        r'''
+        'PHAED': 'made',
+        'KWREU': '{^y}',
+
+        :spaces_after
+        PHAED/KWREU  'mady '
+        * 'made '
+        '''
+
     def test_after_initial_being(self):
         r'''
         '-B': 'be',
```

Test result:

```
_______________________________ TestsBlackbox.test_orthography_issue_961 ________________________________

bb = <test.test_blackbox.TestsBlackbox object at 0x12352b5c0>

>   new_f = lambda bb: (f(bb), replay(bb, name, test))
E   AssertionError: test_orthography_issue_961
E
E     'PHAED': 'made',
E     'KWREU': '{^y}',
E
E     :spaces_after
E     PHAED/KWREU  'mady '
E   > * 'made '
E
E      'mad '
E   != 'made '

plover_build_utils/testing.py:118: AssertionError
============================ 1 failed, 765 passed, 2 xfailed in 6.60 seconds ============================
```

**Comment by benoit-pierre** [comment]
Fixed in #965.
